src/net.cpp

```diff
@@ -1781,9 +1817,14 @@ void CConnman::ThreadOpenConnections(const std::vector<std::string> connect)
         //  * Only make a feeler connection once every few minutes.
         //
         bool fFeeler = false;
-        if (nOutbound >= nMaxOutbound) {
+
+        bool extra_outbound = GetTryNewOutboundPeer();
+
+        int desired_outbound = nMaxOutbound + (extra_outbound ? nMaxExtraOutbound : 0);
+
+        if (nOutbound >= desired_outbound) {
```

Comment:
Not sure if this comment was made before we chatted offline, but it is possible to have both a feeler and an "extra" outbound peer simultaneously (just not in the same loop iteration, but that's nonsensical anyway since we only initiate a single connection in a single loop iteration, regardless).